src/app/model-download/components/model-download-component/model-download-component.component.ts

```diff
@@ -0,0 +1,172 @@
+import { WeakSourceApolloService } from 'src/app/base/services/weak-source/weak-source-apollo.service';
+import { dateAsUTCDate } from 'src/app/util/helper-functions';
+
+@Component({
+  selector: 'kern-model-download-component',
+  templateUrl: './model-download-component.component.html',
+  styleUrls: ['./model-download-component.component.scss']
+})
+export class ModelDownloadComponentComponent implements OnInit {
+
+  project$: any;
+  projectQuery$: any;
+  downloadedModelsList$: any;
+  downloadedModelsQuery$: any;
+  projectId: string;
+  form: FormGroup;
+  models: any[] = [];
+  isManaged: boolean = false;
+  currentModelHandle: any;
+  downloadedModels: any[] = [];
+  subscriptions$: Subscription[] = [];
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private projectApolloService: ProjectApolloService,
+    private routeService: RouteService,
+    private informationSourceApolloService: WeakSourceApolloService,
+    private formBuilder: FormBuilder
+  ) { }
+
+  ngOnInit(): void {
+    this.initForm();
+    this.routeService.updateActivatedRoute(this.activatedRoute);
+    this.projectId = this.activatedRoute.parent.snapshot.paramMap.get('projectId');
+    [this.projectQuery$, this.project$] = this.projectApolloService.getProjectByIdQuery(this.projectId);
+    [this.downloadedModelsQuery$, this.downloadedModelsList$] = this.informationSourceApolloService.getModelProviderInfo();
+    this.prepareModels();
+    this.prepareZeroShotRecommendations(this.projectId);
+    this.isManaged = ConfigManager.getIsManaged();
+
+    this.subscriptions$.push(
+      this.downloadedModelsList$.subscribe((downloadedModels) => this.downloadedModels = downloadedModels));
+
+    NotificationService.subscribeToNotification(this, {
+      projectId: this.projectId,
+      whitelist: ['model_provider_download'],
+      func: this.handleWebsocketNotification
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions$.forEach((subscription) => subscription.unsubscribe());
+    NotificationService.unsubscribeFromNotification(this, this.projectId)
+  }
+
+  initForm() {
+    this.form = this.formBuilder.group({
+      name: ['', Validators.required]
+    })
+  }
+
+  deleteModel(name: string, revision: string) {
+    this.subscriptions$.push(
+      this.informationSourceApolloService
+      .deleteModel(name,revision)
+      .pipe(first()).subscribe()
+    );
+  }
+
+  downloadModel() {
+    if(this.form.invalid) return ;
+    this.subscriptions$.push(
+      this.informationSourceApolloService
+      .downloadModel(this.projectId, this.form.get('name').value)
+      .pipe(first()).subscribe()
+    );
+  }
+
+  parseUTC(utc: any) {
+    const milliseconds = +utc * 1000;
+    const utcDate = dateAsUTCDate(new Date(milliseconds));
+    return utcDate.toLocaleString();
+  }
+
+  prepareModels() {
+    this.subscriptions$.push(this.projectApolloService.getRecomendedEncodersForEmbeddings(this.projectId)
```

Comment:
not sure why this is used without pipe(first()) since the recommendations are hardcoded and shouldn't change over time